src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -0,0 +1,152 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRBaseDevice.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class MTRDeviceController;
+@class MTRAsyncCallbackWorkQueue;
+
+@protocol MTRDeviceSubscriptionDelegate;
+
+@interface MTRDevice : NSObject
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Directly instantiate a MTRDevice with a MTRDeviceController as a shim.
+ *
+ * All device-specific information would be stored on the device controller, and
+ * retrieved when performing actions using a combination of MTRBaseDevice
+ * and MTRAsyncCallbackQueue.
+ */
++ (instancetype)deviceWithDeviceID:(uint64_t)deviceID deviceController:(MTRDeviceController *)deviceController;
+
+/**
+ * Subscribe to receive attribute reports for everything (all endpoints, all
+ * clusters, all attributes, all events) on the device.
+ *
+ * The subscriber provides a delegate object conforming to MTRDeviceSubscriptionDelegate
+ */
+- (void)subscribeWithDelegate:(id<MTRDeviceSubscriptionDelegate>)delegate
+                        queue:(dispatch_queue_t)queue
+                  minInterval:(uint16_t)minInterval
+                  maxInterval:(uint16_t)maxInterval
+                       params:(MTRSubscribeParams * _Nullable)params;
+
+/**
+ * Read attribute in a designated attribute path
+ *
+ * @return a data-value dictionary of the attribute as described in MTRDeviceResponseHandler
+ */
+- (NSDictionary<NSString *, id> *)readAttributeWithEndpointId:(NSNumber * _Nullable)endpointId
+                                                    clusterId:(NSNumber * _Nullable)clusterId
+                                                  attributeId:(NSNumber * _Nullable)attributeId
+                                                       params:(MTRReadParams * _Nullable)params;
+
+/**
+ * Write to attribute in a designated attribute path
+ *
+ * @param value       A data-value NSDictionary object as described in
+ *                    MTRDeviceResponseHandler.
+ *
+ * @param expectedValueIntervalMs  interval that the write value is assumed to hold true before actual interaction happens. This
+ * value will be clamped to timeoutMs.
+ *
+ * @param timeoutMs   timeout in milliseconds for timed write, or nil.
+ *
+ *                    Received values are an NSArray object with response-value element as described in
+ *                    readAttributeWithEndpointId:clusterId:attributeId:clientQueue:completion:.
```

Comment:
This is left over from the copy/paste. This API removed the completion handler because it expects the write to succeed, and any further changes to that expectation is to be delivered via the subscription callbacks.

And so this text will be removed.